exporters/trace/instana/src/main/java/io/opencensus/exporter/trace/instana/InstanaExporterHandler.java

```diff
@@ -0,0 +1,224 @@
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.logging.Logger;
+
+/*
+ * Exports to an Instana agent acting as proxy to the Instana backend (and handling authentication)
+ * Uses the Trace SDK documented:
+ * https://github.com/instana/instana-java-sdk#instana-trace-webservice
+ *
+ * Currently does a blocking export using HttpUrlConnection.
+ * Also uses a StringBuilder to build JSON.
+ * Both can be improved should 3rd party library usage not be a concern.
+ *
+ * Major TODO is the limitation of Instana to only suport 64bit trace ids, which will be resolved.
+ * Until then it is crossing fingers and treating it as 50% sampler :).
+ */
+final class InstanaExporterHandler extends SpanExporter.Handler {
+
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final Sampler probabilitySpampler = Samplers.probabilitySampler(0.0001);
+  static final Logger logger = Logger.getLogger(InstanaExporterHandler.class.getName());
+
+  private static final Function<Object, String> RETURN_STRING =
+      new Function<Object, String>() {
+        @Override
+        public String apply(Object input) {
+          return input.toString();
+        }
+      };
+
+  private final URL agentEndpoint;
+
+  InstanaExporterHandler(URL agentEndpoint) {
+    this.agentEndpoint = agentEndpoint;
+  }
+
+  private static String encodeTraceId(TraceId traceId) {
+    return BaseEncoding.base16().lowerCase().encode(traceId.getBytes(), 0, 8);
+  }
+
+  private static String encodeSpanId(SpanId spanId) {
+    return BaseEncoding.base16().lowerCase().encode(spanId.getBytes());
+  }
+
+  private static String toSpanType(SpanData spanData) {
+    if (spanData.getParentSpanId() == null || Boolean.TRUE.equals(spanData.getHasRemoteParent())) {
+      return "ENTRY";
+    }
+
+    // This is a hack because the v2 API does not have SpanKind. When switch to v2 this will be
+    // fixed.
+    if (spanData.getName().startsWith("Sent.")) {
+      return "EXIT";
+    }
+
+    return "INTERMEDIATE";
+  }
+
+  private static long toMillis(Timestamp timestamp) {
+    return SECONDS.toMillis(timestamp.getSeconds()) + NANOSECONDS.toMillis(timestamp.getNanos());
+  }
+
+  private static long toMillis(Timestamp start, Timestamp end) {
+    Duration duration = end.subtractTimestamp(start);
+    return SECONDS.toMillis(duration.getSeconds()) + NANOSECONDS.toMillis(duration.getNanos());
+  }
+
+  private static String attributeValueToString(AttributeValue attributeValue) {
+    return attributeValue.match(
+        RETURN_STRING, RETURN_STRING, RETURN_STRING, Functions.<String>returnNull());
+  }
+
+  static String convertToJson(Collection<SpanData> spanDataList) {
+    StringBuilder sb = new StringBuilder();
+    sb.append('[');
+    for (SpanData span : spanDataList) {
+      if (span.getStatus() == null) {
+        // TODO: Instana expects completed spans. Will those come in again?
```

Comment:
I could not find any explanation of why a incomplete span is ever sent to exporter, and if it is sent again when complete

Reply:
Null span status means running span and that should not happen in exportation, since only ended spans will be added to the queue of span exporter. 

However, a check here would be good. We have [similar logic](https://github.com/census-instrumentation/opencensus-java/blob/3c74cea59b7c50c65e50401305e7f7feab614a90/impl_core/src/main/java/io/opencensus/implcore/trace/export/SampledSpanStoreImpl.java#L180) to ensure the library does not crash

Reply:
ok, thanks, will leave it in as a safety anyways.